dateutil/test/test_tz.py

```diff
@@ -1077,6 +1077,12 @@ def testGettzCacheTzLocal(self):
         local2 = tz.gettz()
 
         assert local1 is not local2
+    
+    def test_gettz_same_result_for_none_and_empty_string(self):
```

Comment:
I think we want to assert something specific about the result - that it's not None and that it is a local time.

I'd refactor this as a parameterized test parameterizing `args` over `() and ("",)`, then check that `tz.gettz(*args)` returns a `tz.tzlocal` or whatever it's supposed to return.

Reply:
Since it's platform dependent I was not sure if there is a good way to get local time other than using `gettz`. I actually had the following test:
```python
@pytest.mark.skipif(IS_WIN, reason='requires Unix')	
def test_gettz_returns_local_time(self):	
    with TZEnvContext('/etc/localtime'):	
        assert tz.gettz() == tz.tzfile('/etc/localtime')
```
However, since not all unix systems will have a tzfile at this location, this test is not really reliable. I don't really understand what `tz.tzlocal` is/does, so I was not sure if there is a way to use it here.

Reply:
I also thought `tz.gettz()` returning local time would hopefully have been tested already, although I did not find a test for it.

Reply:
Ah, that's a good point. We can start with just checking that it's not `None` and that it's the same as `tz.gettz()`.

I think we can add a test for the "is it the local time" behavior with some variation on this:

```python
def test_gettz_returns_local(self, example_datetime):
    dt_act = example_datetime.replace(tzinfo=tz.gettz())
    if six.PY2:
        dt_exp = example_datetime.replace(tzinfo=tz.tzlocal())
    else:
        dt_exp = example_datetime.astimezone()

    assert dt_act == dt_exp
    assert dt_act.tzname() == dt_exp.tzname()
    assert dt_act.utcoffset() == dt_exp.utcoffset()
    assert dt_act.dst() == dt_exp.dst()
```

We can either make this a property test and have `example_datetime` be generated by a `datetime` strategy or we can just parametrize over `example_datetime` and pick a few random datetimes and call it a day.